spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -81,6 +110,20 @@ interface UpgradeTimeout {
 
 At least one of the `timeoutHeight` or `timeoutTimestamp` MUST be non-zero.
 
+#### `Upgrade`
+
+The upgrade type will represent a particular upgrade attempt on a channel end.
+
+```typescript
+interface Upgrade {
+    fields: UpgradeFields
+    timeout: UpgradeTimeout
+    lastPacketSent: uint64
+}
+```
+
+The upgrade contains the proposed upgrade for the channel end on the executing chain, the timeout for the upgrade attempt, and the last packet send sequence for the channel. The `lastPacketSent` allows the counterparty to know which packets need to be flushed before the channel can reopen with the newly negotiated parameters.
```

Comment:
Could you elaborate a bit more how `lastPacketSent` should be used in the counterparty? I am wondering now if this is needed, since a chain can take care of flushing all the packets that it has sent, but doesn't need to worry about the packets sent by the counterparty. If the chain doesn't allow sending new packets, it should not be possible for the counterparty to receive packets with a sequence higher than `lastPacketSent`, right?